cognitive_architecture/database/create_database.py

```diff
@@ -1,63 +1,135 @@
-    cursor = connection.cursor()
-    cursor.execute(f"DROP DATABASE IF EXISTS {db_name}")
-    cursor.close()
+class DatabaseManager:
+    """Manages database creation, deletion, and table initialization."""
+    def __init__(self, config: DatabaseConfig):
+        """Initialize the DatabaseManager with a given configuration."""
+        self.config = config
+        self.engine = create_async_engine(config.get_sqlalchemy_database_url(), echo=True)
+        self.db_type = config.db_type
 
+    @asynccontextmanager
+    async def get_connection(self):
+        """Initialize the DatabaseManager with a given configuration."""
+        if self.db_type in ["sqlite", "duckdb"]:
+            # For SQLite and DuckDB, the engine itself manages connections
+            yield self.engine
+        else:
+            async with self.engine.connect() as connection:
+                yield connection
 
+    async def database_exists(self, db_name):
+        """Check if a database exists."""
+        if self.db_type in ["sqlite", "duckdb"]:
+            # For SQLite and DuckDB, check if the database file exists
+            return os.path.exists(db_name)
+        else:
+            query = text(f"SELECT 1 FROM pg_database WHERE datname='{db_name}'")
+            async with self.get_connection() as connection:
+                result = await connection.execute(query)
+                return await result.fetchone() is not None
 
-def create_tables(engine):
-    Base.metadata.create_all(bind = engine)
+    async def create_database(self, db_name):
+        """Create a new database."""
+        if self.db_type not in ["sqlite", "duckdb"]:
+            # For databases like PostgreSQL, create the database explicitly
+            async with self.get_connection() as connection:
+                await connection.execute(text(f"CREATE DATABASE {db_name}"))
 
-if __name__ == "__main__":
-    host = os.environ.get('POSTGRES_HOST')
-    username = os.environ.get('POSTGRES_USER')
-    password = os.environ.get('POSTGRES_PASSWORD')
-    database_name = os.environ.get('POSTGRES_DB')
+    async def drop_database(self, db_name):
+        """Drop an existing database."""
+        if self.db_type in ["sqlite", "duckdb"]:
+            # For SQLite and DuckDB, simply remove the database file
+            os.remove(db_name)
+        else:
+            async with self.get_connection() as connection:
+                await connection.execute(text(f"DROP DATABASE IF EXISTS {db_name}"))
+
+    async def create_tables(self):
+        """Create tables based on the SQLAlchemy Base metadata."""
+        async with self.engine.begin() as conn:
+            await conn.run_sync(Base.metadata.create_all)
 
-    engine = create_admin_engine(username, password, host, database_name)
-    connection = engine.connect()
+if __name__ == "__main__":
+    async def main():
+        """Runs as a part of startup docker scripts to create the database and tables."""
```

Comment:
Why not move this code away from this file and just call the functions from that other file?
This file can be then called `databaseManager.py` for example.
Also not sure we need class, functions are fine as well.

Reply:
Refactored